### PR TITLE
Makefile: format all packages in GOFILES

### PR DESCRIPTION
**Pull request by aalemayhu**
Closes: #1445 (Run gofmt as part of the regular build)
Signed-off-by: Alexander Alemayhu <alexander@alemayhu.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -7,6 +7,7 @@ GOLANGVERSION = $(shell go version 2>/dev/null | grep -Eo '(go[0-9].[0-9])')
 GOTEST_OPTS = -test.v -check.v
 
 all: $(SUBDIRS)
+	for pkg in $(GOFILES); do go fmt $$pkg; done
```
Comment (aanm) [inline Makefile]: I think you meant `gofmt -w $$pkg`

**Review by aanm — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (aalemayhu) [reply Makefile]: No, see help page
```
go help fmt
usage: go fmt [-n] [-x] [packages]

Fmt runs the command 'gofmt -l -w' on the packages named
by the import paths.  It prints the names of the files that are modified.
```

**Comment by tgraf** [comment]
Can you provide an example output of how it looks like if there is a formatting error?

**Review by aanm — APPROVED** [review]

**Comment by aalemayhu** [comment]
@tgraf If there is a formatting error you won't see any output.  It would just be reformatted. Considering that this is running right after building where it should not reach reformatting if there is a build error. So seems safe to assume the tooling can reformat in place. Let me know what you think, alternative we can change it if you don't like this.

@aanm Thanks for approving.

**Comment by ianvernon** [comment]
Say someone runs a build with unformatted Go code - this will run `go fmt` on the build slaves, but is it possible that the unformatted code gets merged? I.e., are the changes made in the workspace in Jenkins made reflected in master after a merge is done?

**Comment by tgraf** [comment]
> @tgraf If there is a formatting error you won't see any output. It would just be reformatted. Considering that this is running right after building where it should not reach reformatting if there is a build error. So seems safe to assume the tooling can reformat in place. Let me know what you think, alternative we can change it if you don't like this.

I think failing with the diff will be better. Replacing requires the editor to reload the files. Some editors will do this automatically, most won't.

File: Makefile [hunk 2]
```diff
@@ -7,6 +7,7 @@ GOLANGVERSION = $(shell go version 2>/dev/null | grep -Eo '(go[0-9].[0-9])')
 GOTEST_OPTS = -test.v -check.v
 
 all: $(SUBDIRS)
+	-tests/00-fmt.sh
```
Comment (tgraf) [inline Makefile]: We should run `gofmt` before we compile. Otherwise we have to wait for the compile to be complete every time.

**Review by tgraf — CHANGES_REQUESTED** [review]
The output is great otherwise:
```
make[1]: Leaving directory '/home/tgraf/dev/go/src/github.com/cilium/cilium/monitor'
tests/00-fmt.sh
Unformatted Go source code:
./daemon/daemon.go
diff ./daemon/daemon.go gofmt/./daemon/daemon.go
--- /tmp/gofmt043574680	2017-08-30 03:20:01.370900674 +0200
+++ /tmp/gofmt253938199	2017-08-30 03:20:01.370900674 +0200
@@ -83,7 +83,7 @@
 	initArgLib int = iota
 	initArgRundir
 	initArgIPv6Router
-	   initArgIPv4NodeIP
+	initArgIPv4NodeIP
 	initArgIPv6NodeIP
 	initArgIPv4Range
 	initArgIPv6Range
Makefile:10: recipe for target 'all' failed
make: [all] Error 1 (ignored)
```

**Review by tgraf — APPROVED** [review]

**Pull request merged**